src/unix/tcp.c

```diff
@@ -340,7 +340,9 @@ int uv__tcp_connect(uv_connect_t* req,
   uv__queue_init(&req->queue);
   handle->connect_req = req;
 
-  uv__io_start(handle->loop, &handle->io_watcher, POLLOUT);
+  err = uv__io_start(handle->loop, &handle->io_watcher, POLLOUT);
+  if (err)
+    return err;
```

Comment:
The uv__req_init call a few lines up increases `loop->active_reqs.count`. This return should either adjust it back with uv__req_unregister (okay), or call uv__io_start before uv__req_init (better.)

Reply:
Roger that.